src/impl/vamp/collision/validity.hh

```diff
@@ -252,18 +252,39 @@ namespace vamp
     inline constexpr auto attachment_environment_collision(const collision::Environment<DataT> &e) noexcept
         -> bool
     {
-        for (const auto &s : e.attachments->posed_spheres)
-        {
-            // HACK: The radius needs to be a float, and unfortunately the spheres assume homogeneous
-            // DataT for storage
-            // TODO: Fix the sphere representation to allow to store float radii even with vector
-            // centers
-            if (sphere_environment_in_collision(e, s.x, s.y, s.z, s.r[{0, 0}]))
+        for (const auto &eef_attachment: e.eef_attachments){
+            for (const auto &s : eef_attachment.attachment.posed_spheres)
             {
-                return true;
+                // HACK: The radius needs to be a float, and unfortunately the spheres assume homogeneous
+                // DataT for storage
+                // TODO: Fix the sphere representation to allow to store float radii even with vector
+                // centers
+                if (sphere_environment_in_collision(e, s.x, s.y, s.z, s.r[{0, 0}]))
+                {
+                    return true;
+                }
             }
+
+            // now treat the other attachments also as potential collision objects
+            // This is probably expensive, find a more efficient way. 
```

Comment:
Attachment collision checking is comparatively cheap; I think it is ok to stay.
Maybe adding a coarse phase check would be an improvement to perf but generally I think it is not going to be a big issue.